torch/csrc/autograd/engine.cpp

```diff
@@ -376,6 +371,7 @@ static variable_list call_function(FunctionTask& task) {
   checkpoint_valid = prev_checkpoint_valid_state;
 
   if(has_post_hooks){
+    // NOLINTNEXTLINE(bugprone-use-after-move)
```

Comment:
Maybe it would be simpler to move the `return` below to an `else` branch? That should silence the warning